docs/migration_to_1.0.0.md

```diff
@@ -0,0 +1,72 @@
+# Migrating from Wasmer 0.x to Wasmer 1.0.0
+
+Wasmer 1.0.0 is currently in alpha and is our primary focus. This document will
+describe the differences between 0.x Wasmer and Wasmer 1.0.0 and provide examples
+to make migrating to the new API as simple as possible.
+
+Some features are still under development during the alpha of Wasmer 1.0.0. This document
+will aim to make clear what these features are.
+
+## Table of Contents
+
+- Rationale for changes in 1.0.0
+- How to use Wasmer 1.0.0
+- Project structure
+- TODO: specific differences
+
+## Rationale for changes in 1.0.0
+
+TODO: explain why 1.0.0 exists
+
+## How to use Wasmer 1.0.0
+
+### Installing Wasmer
+
+See [wasmer.io] for installation instructions.
+
+If you already have wasmer installed, run `wasmer self-update`.
+
+Install the latest versions of wasmer with [wasmer-nightly].
+
```

Comment:
Based on NEAR suggestion, we should also add a `## Errors` section.

Reply:
If we had something about commons errors and how to fix them, I think the best place would be the documentation.

Thoughts?